app/controllers/posts_controller.rb

```diff
@@ -4,7 +4,29 @@ def index
     @posts = Post.includes(:user).where(user: params[:user_id])
   end
 
+  def new
+    @post = Post.new
+  end
+
   def show
+    @user = User.find(params[:user_id])
     @post = Post.includes(:user, comments: [:user]).find(params[:id])
   end
+
+  def create
+    user = User.find(params[:user_id])
+    created_post = Post.new(params.require(:post).permit(:title, :text))
+    created_post.commentscounter = 0
+    created_post.likescounter = 0
+    created_post.authorId = user.id
+
+    if created_post.save
```

Comment:
Hi @codepantha, to create a post I implemented the strong params as seen on my line 18. Can you please show me what else to implement?

Reply:
- [ ] Yes, you did. I meant to implement it using a private method like this 👇 
```
private

  def post_params
    params.require(:post).permit(:title, :text)
  end
```
so that your controller is cleaner.